happybase/pool.py

```diff
@@ -90,6 +90,33 @@ def _return_connection(self, connection):
         """Return a connection to the pool."""
         self._queue.put(connection)
 
+    def close_connections(self, timeout=None):
+        """
+        Attempts to politely close all connections in the pool.
+        Waits for used connections to become available or until the
+        timeout is exceeded.
+
+        :param int timeout: number of seconds to wait for a connection to
+                            become available (optional)
+        :return: None
+        """
+        if timeout:
+            if not isinstance(timeout, int):
+                raise TypeError("close_connections 'timeout' arg must be an integer")
+
+            if not timeout > 0:
+                raise ValueError("close_connections 'timeout' arg must be greater than zero")
+
+        while not self._queue.empty():
+            try:
+                conn = self._queue.get(block=True, timeout=timeout)
+                conn.close()
+                conn = None
+            except Queue.Empty:
+                raise NoConnectionsAvailable(
+                    "Closing connections failed: No connection available from "
+                    "pool within specified timeout of {}".format(timeout))
```

Comment:
I don't have a good way to non-destructively drain the queue. the queue is an internal representation, but this code doesn't change that it only allows the caller to 'politely' terminate the connections when they see fit.
perhaps it's named badly? a better name might be "terminate_connections" the current name implies a different behavior.
thoughts? ... I feel like I'm missing your point a bit.